common/djangoapps/student/views.py

```diff
@@ -609,10 +609,11 @@ def dashboard(request):
         and has_access(request.user, 'view_courseware_with_prerequisites', enrollment.course_overview)
     )
 
-    # Get any programs associated with courses being displayed.
-    # This is passed along in the template context to allow rendering of
-    # program-related information on the dashboard.
-    course_programs = _get_course_programs(user, [enrollment.course_id for enrollment in course_enrollments])
+    # Find programs associated with courses being displayed. This information
+    # is passed in the template context to allow rendering of program-related
+    # information on the dashboard.
+    programs, __ = programs_utils.attach_program_detail_url(programs_utils.get_programs(user))
```

Comment:
Why use this function instead of `ProgramProgressMeter.engaged_programs`? At least the `engaged_programs` reads better